scipy/_lib/_elementwise_iterative_method.py

```diff
@@ -313,28 +314,28 @@ def _update_active(work, res, res_work_pairs, active, mask, preserve_shape, xp):
     if mask is not None:
         if preserve_shape:
             active_mask = xp.zeros_like(mask)
-            active_mask[active] = 1
+            active_mask = xpx.at(active_mask)[active].set(True)
             active_mask = active_mask & mask
             for key, val in update_dict.items():
                 try:
-                    res[key][active_mask] = val[active_mask]
+                    res[key] = xpx.at(res[key])[active_mask].set(val[active_mask])
                 except (IndexError, TypeError, KeyError):
-                    res[key][active_mask] = val
+                    res[key] = xpx.at(res[key])[active_mask].set(val)
```

Comment:
This is going to fail in jax.jit, and the exception filter is so wide that it _might_ quietly suppress the jax tracer error depending on what it subclasses. Unsure what the original intent was?
Is `val` sometimes a scalar? In which case this should be

```python
if getattr(val, ndim, 0) == 0:
    res[key] = xpx.at(res[key])[active_mask].set(val)
else:
    assert val.shape == res[key].shape
    res[key] = xp.where(active_mask, val, res[key])

Reply:
I thought this might come up. My plan was to fix this `try`-`except` business right after this PR. I happened to have already done this a few days ago, and I didn't want to cause merge conflicts.

Can we judge this on the basis of converting in-place operations to `xpx.at`, and tackle that separately?

Reply:
works for me